ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -358,27 +351,23 @@ - (BOOL)mapView:(RCTMapboxGL *)mapView annotationCanShowCallout:(id <MGLAnnotati
     }
 }
 
-- (void)selectAnnotationAnimated:(NSUInteger)annotationInArray
+- (void)selectAnnotationAnimated:(NSString*)selectedIdentifier
 {
-    if (annotationInArray >= [_annotations count]) {
-        RCTLogError(@"Could not find annotation in array");
-        return;
-    }
-    if ([_annotations count] != 0) {
-        [_map selectAnnotation:_annotations[annotationInArray] animated:YES];
-    }
+    [_map selectAnnotation:[_annotations objectForKey:selectedIdentifier] animated:YES];
 }
 
-- (void)removeAnnotation:(NSUInteger)annotationInArray
+- (void)removeAnnotation:(NSString*)selectedIdentifier
 {
-    if (annotationInArray >= [_annotations count]) {
-        RCTLogError(@"Could not find annotation in array");
-        return;
-    }
-    if ([_annotations count] != 0) {
-        [_map removeAnnotation:_annotations[annotationInArray]];
-        [_annotations removeObjectAtIndex:annotationInArray];
+    [_map removeAnnotation:[_annotations objectForKey:selectedIdentifier]];
+    [_annotations removeObjectForKey:[_annotations objectForKey:selectedIdentifier]];
+}
+
+- (void)removeAllAnnotations
+{
+    for(id key in _annotations) {
```

Comment:
You can replace this loop with a single call to `[_map removeAnnotations:_map.annotations]`.